packages/flutter_markdown/test/custom_syntax_test.dart

```diff
@@ -58,6 +59,30 @@ void defineTests() {
         expect(span.recognizer.runtimeType, equals(TapGestureRecognizer));
       },
     );
+
+    testWidgets(
+      'replace word "container" with container widget',
+      (WidgetTester tester) async {
+        await tester.pumpWidget(
+          boilerplate(
+            Markdown(
+              data: 'container is a widget that allows to customize its child',
+              extensionSet: md.ExtensionSet.none,
+              inlineSyntaxes: <md.InlineSyntax>[ContainerSyntax()],
+              builders: <String, MarkdownElementBuilder>{
+                'container': ContainerBuilder(),
+              },
+            ),
+          ),
+        );
+
+        final RichText textWidget = tester.widget(find.byType(RichText));
+        final TextSpan span =
+            (textWidget.text as TextSpan).children![0] as TextSpan;
+        final WidgetSpan widgetSpan = span.children![0] as WidgetSpan;
+        expect(widgetSpan.child, isInstanceOf<Container>());
```

Comment:
I don't know this codebase all that well, but I'd like to see comprehensive tests. This test feels like it is testing what the code does with the singular case that is currently known broken.

Can we add more cases to the test coverage that demonstrate various forms of usage to make sure everything stays sane and we don't have to reverse this change out in three months?